src/Illuminate/Auth/Access/Gate.php

```diff
@@ -148,6 +150,38 @@ public function denies($ability, $arguments = [])
      */
     public function check($ability, $arguments = [])
```

Comment:
@garygreen the overhead of creating a simple grant object is negligible. I switched it out to what you proposed.